packages/playwright/src/runner/watchMode.ts

```diff
@@ -251,12 +252,37 @@ export async function runWatchModeLoop(configLocation: ConfigLocation, initialOp
   return result === 'passed' ? teardown.status : result;
 }
 
+function readKeyPress(handler: (text: string, key: any) => void): Disposable {
+  const rl = readline.createInterface({ input: process.stdin, escapeCodeTimeout: 50 });
+  readline.emitKeypressEvents(process.stdin, rl);
+  if (process.stdin.isTTY)
+    process.stdin.setRawMode(true);
+
+  process.stdin.on('keypress', handler);
+
+  return {
```

Comment:
returning the callback effect-style would also work. We only use Disposable when we interoperate with vscode.

Reply:
done in https://github.com/microsoft/playwright/pull/32584/commits/f0de6ccd81ff07f9e947625839b3335e0da3ba4a